pinot-segment-local/src/main/java/org/apache/pinot/segment/local/function/GroovyFunctionEvaluator.java

```diff
@@ -83,7 +83,12 @@ public GroovyFunctionEvaluator(String closure) {
     _numArguments = _arguments.size();
     _binding = new Binding();
     String scriptText = matcher.group(SCRIPT_GROUP_NAME);
-    _script = createSafeShell(_binding).parse(scriptText);
+    try {
+      _script = createSafeShell(_binding).parse(scriptText);
+    } catch (Exception e) {
+      LOGGER.error("Groovy compilation error: " + closure, e);
```

Comment:
We don't usually log error if exception is thrown out, especially for user error